pkg/transparentproxy/transparentproxy_experimental_test.go

```diff
@@ -0,0 +1,19 @@
+package transparentproxy_test
+
+import (
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	"github.com/kumahq/kuma/pkg/transparentproxy"
+)
+
+var _ = Describe("ShouldEnableIPv6", func() {
```

Comment:
It's more complicated to create test that checks other cases. It might be flaky depending on the system you run it on. `ShouldEnableIPv6` check network interfaces and return true/false depending on the system configuration.